src/generate/generic.rs

```diff
@@ -104,17 +104,33 @@ pub trait FieldSpec: Sized {
 /// Marker for fields with fixed values
 pub trait IsEnum: FieldSpec {}
 
+#[doc(hidden)]
+pub trait FromBits<U> {
+    unsafe fn from_bits(b: U) -> Self;
+}
+
+#[doc(hidden)]
+pub trait ToBits<U> {
+    fn to_bits(&self) -> U;
+}
```

Comment:
If these traits are not intended to be used externally, maybe we can implement `Readable` without the extra layer of trait resolution?

I'm thinking along the lines of creating a helper macro to implement the `Readable` trait on the needed types.

Reply:
I forgot about `inline`.
Now it is 7.94 s